File: src/main/java/skywolf46/extrautility/events/interaction/PlayerRightClickEvent.java
package skywolf46.extrautility.events.interaction;

import org.bukkit.entity.Player;
import org.bukkit.event.HandlerList;
import org.bukkit.inventory.ItemStack;
import skywolf46.extrautility.events.abstraction.AbstractPlayerItemEvent;

public class PlayerRightClickEvent extends AbstractPlayerItemEvent {
    private static HandlerList handlerList = new HandlerList();
    private boolean isOffHanded;

    public PlayerRightClickEvent(Player who, boolean isOffHanded) {
        super(who);
        this.isOffHanded = isOffHanded;
    }

    @Override
    public HandlerList getHandlers() {
        return handlerList;
    }

    public static HandlerList getHandlerList() {
        return handlerList;
    }

    @Override
    public ItemStack getItemInHand() {
        return isOffHanded ? getPlayer().getEquipment().getItemInOffHand() : getPlayer().getEquipment().getItemInMainHand();
    }

    @Override
    public void setItemInHand(ItemStack ita) {
        if(isOffHanded)
            getPlayer().getEquipment().setItemInOffHand(ita);
        else
            getPlayer().getEquipment().setItemInMainHand(ita);
    }
}

File: minecraft/src/main/kotlin/skywolf46/extrautility/test/events/interaction/PlayerRightClickEvent.kt
package skywolf46.extrautility.test.events.interaction

import org.bukkit.entity.Player
import org.bukkit.event.HandlerList
import org.bukkit.event.player.PlayerInteractEvent
import skywolf46.extrautilitytest.test.events.abstraction.AbstractPlayerItemEvent


class PlayerRightClickEvent(
    ev: PlayerInteractEvent,
    who: Player?,
    val isBlockInteraction: Boolean,
    val isOffHanded: Boolean
) :
    AbstractPlayerItemEvent(ev, who) {
    override fun getHandlers(): HandlerList {
        return _handle
    }

    companion object {

        internal val _handle = HandlerList()

        @JvmStatic
        fun getHandlerList(): HandlerList {
            return _handle
        }
    }
}
